src/core/api/link_api.cpp

```diff
@@ -69,7 +69,7 @@ const otExtAddress *otLinkGetExtendedAddress(otInstance *aInstance)
 {
     Instance &instance = *static_cast<Instance *>(aInstance);
 
-    return reinterpret_cast<const otExtAddress *>(instance.GetThreadNetif().GetMac().GetExtAddress());
+    return &static_cast<const otExtAddress &>(instance.GetThreadNetif().GetMac().GetExtAddress());
```

Comment:
Can we possibly remove the `static_cast<>` here and just return `&instance.GetThreadNetif().GetMac().GetExtAddress()` instead? 
I think pointer to a sub-class instance `Mac::ExtAddress` should be automatically a valid base class `otExtAddress` pointer.